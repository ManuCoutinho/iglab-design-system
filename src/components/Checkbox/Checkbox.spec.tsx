import { screen, render, fireEvent } from '@testing-library/react'
import { Checkbox } from '.'

describe('<Checkbox/>', () => {
  it('should render Checkbox correctly ', () => {
    render(<Checkbox />)
    expect(screen.getByRole('checkbox')).toBeInTheDocument()
    expect(screen.getByRole('checkbox')).toHaveAttribute(
      'aria-checked',
      'false'
    )
  })
  it('should render Checkbox with checked attribute ', () => {
    render(<Checkbox />)
    const checkbox = screen.getByRole('checkbox')
    fireEvent.click(checkbox)
    expect(checkbox).toHaveAttribute('aria-checked', 'true')
  })
  it('match to snapshot', () => {
    const { container } = render(<Checkbox />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
