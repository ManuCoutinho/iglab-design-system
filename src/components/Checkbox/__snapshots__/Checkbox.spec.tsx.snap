// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> match to snapshot 1`] = `
<button
  aria-checked="false"
  class="w-6 h-6 p-[2px] bg-gray-800 rounded transition focus-within:ring focus-within:ring-cyan-500 focus-within:outline-cyan-300"
  data-state="unchecked"
  role="checkbox"
  title="checkbox"
  type="button"
  value="on"
/>
`;
